### PR TITLE
Feature/verify page

### PR DESCRIPTION
**Pull request by Khushboo1008**

### CODE REVIEW

File: src/containers/verify/components/Login.tsx [hunk 1]
```diff
@@ -0,0 +1,71 @@
+import React, { useState } from 'react'
+import bcrypt from 'bcryptjs-react'
+import { useAppDispatch } from '@/redux/hooks'
+import { toast, Toaster } from 'react-hot-toast'
+import { setStep } from '@/redux/verify'
+import { STATIC_PASSWORD } from '@/utils/constants'
+
+// const STATIC_PASSWORD =
```
Comment (codebuster22) [inline src/containers/verify/components/Login.tsx:8]: is this comment required?

File: src/containers/verify/components/Login.tsx [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+//   '$2a$10$hgdaIRpS3h56ZKL9amZQreVDQgJY3T9VR4wv/Y7P3/tvbhEggaVgO'
+
+const Login = () => {
+  const [password, setPassword] = useState('')
+  const dispatch = useAppDispatch()
+
+  const handleLogin = () => {
+    const salt = bcrypt.genSaltSync(10)
+    const hash = bcrypt.hashSync(password, salt)
+    console.log(hash)
```
Comment (codebuster22) [inline src/containers/verify/components/Login.tsx:18]: can you log it properly? Describe what this is logging so we can debug properly.

File: src/containers/verify/components/Login.tsx [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+    console.log(STATIC_PASSWORD)
+    if (bcrypt.compareSync(password, STATIC_PASSWORD)) {
+      console.log('valid')
```
Comment (codebuster22) [inline src/containers/verify/components/Login.tsx:21]: what is valid

File: src/containers/verify/components/Login.tsx [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+      toast('login Successfull', { duration: 3000 })
+      setTimeout(changeStep, 4000)
```
Comment (codebuster22) [inline src/containers/verify/components/Login.tsx:23]: why do we wait for 4 seconds before moving to next step?

File: src/containers/verify/components/Login.tsx [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+    } else {
+      toast('Invalid Password')
+      setPassword('')
+    }
+  }
+
+  const changeStep = () => {
+    dispatch(setStep(1))
```
Comment (codebuster22) [inline src/containers/verify/components/Login.tsx:31]: define steps as String constants or enums. No numbers.
What step are we setting it to?

File: src/containers/verify/components/QrScan.tsx [hunk 6]
```diff
@@ -0,0 +1,153 @@
...
+  const [error, setError] = useState('')
+  const [message, setMessage] = useState('')
+
+  const handleQrCodeData = async (result) => {
+    const qrCodeData = JSON.parse(result)
+    const signerAddress = ethers.utils.verifyMessage(
+      qrCodeData.message,
+      qrCodeData.signature,
+    )
+    console.log(signerAddress)
```
Comment (codebuster22) [inline src/containers/verify/components/QrScan.tsx:25]: what does this log say? How will this help in debugging?

**Review by codebuster22 — CHANGES_REQUESTED** [review]
looks good, but some changes required.

File: src/containers/verify/components/Login.tsx (hunk 2 above)
Comment (angel112) [reply src/containers/verify/components/Login.tsx:18]: wrap it like `console.log({ hash })` and you can name it as well like `console.log({ name: hash })`

Comment (codebuster22) [reply src/containers/verify/components/Login.tsx:18]: I would suggest adding a string description: 
"Login hash: "

File: src/containers/verify/components/Login.tsx [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+
+// const STATIC_PASSWORD =
+//   '$2a$10$hgdaIRpS3h56ZKL9amZQreVDQgJY3T9VR4wv/Y7P3/tvbhEggaVgO'
+
+const Login = () => {
+  const [password, setPassword] = useState('')
+  const dispatch = useAppDispatch()
+
+  const handleLogin = () => {
+    const salt = bcrypt.genSaltSync(10)
```
Comment (codebuster22) [inline src/containers/verify/components/Login.tsx:16]: magic value. Remove it.

File: src/containers/verify/components/QrScan.tsx [hunk 8]
```diff
@@ -0,0 +1,153 @@
...
+      const serverResponse = await sendTokenIdToServer(body)
+      console.log(serverResponse)
+      if (serverResponse) {
+        handleCloseScan()
+        setErrorOcurred(!serverResponse.data.success)
+        setMessage(serverResponse.data.data.message)
+        setShowModal(true)
+      }
+    } else {
+      setErrorOcurred(true)
```
Comment (codebuster22) [inline src/containers/verify/components/QrScan.tsx:48]: What error occurred?

File: src/containers/verify/components/QrScan.tsx [hunk 9]
```diff
@@ -0,0 +1,153 @@
...
+      setError('Owner is not valid')
```
Comment (codebuster22) [inline src/containers/verify/components/QrScan.tsx:49]: Error string should be constant.

Comment (codebuster22) [reply src/containers/verify/components/QrScan.tsx:49]: example
```export const ERROR_OWNER_NOT_VALID = "Owner is not valid";```

File: src/containers/verify/components/QrScan.tsx [hunk 10]
```diff
@@ -0,0 +1,153 @@
...
+    }
+    setLoadingScan(false)
+  }
+
+  const checkIfSignerAddressIsOwnerOfTokenId = async (
+    tokenId,
+    contractAddress,
+    signerAddress,
+  ) => {
+    const id = `ticket-${contractAddress}-${tokenId}`
```
Comment (codebuster22) [inline src/containers/verify/components/QrScan.tsx:60]: this can be converted to pure function. To ensure generation of ticket ID is consistent everywhere

**Review by codebuster22 — CHANGES_REQUESTED** [review]

File: package.json [hunk 11]
```diff
@@ -24,21 +24,25 @@
     "@types/react-dom": "18.0.10",
     "akar-icons": "^1.9.22",
     "axios": "^1.3.2",
+    "bcrypt": "^5.1.0",
```
Comment (codebuster22) [inline package.json:27]: If we are not using, can we remove it?

**Review by codebuster22 — CHANGES_REQUESTED** [review]

File: src/containers/verify/components/Login.tsx (hunk 1 above)
Comment (codebuster22) [reply src/containers/verify/components/Login.tsx]: can this comment be removed?

File: src/containers/verify/components/Login.tsx (hunk 7 above)
Comment (codebuster22) [reply src/containers/verify/components/Login.tsx]: still using magic value?

**Review by codebuster22 — CHANGES_REQUESTED** [review]

**Review by codebuster22 — APPROVED** [review]
lgtm

**Pull request merged**